src/model/model.nit

```diff
@@ -2572,6 +2572,12 @@ end
 class MClassKind
 	redef var to_s
 
+	# Can a class of kind `self` define a membership predicate?
+	var can_customize_isa: Bool
+
+	# Can a class of kind `self` define a constructor?
+	var can_init: Bool
```

Comment:
For now, yes. But, they are completely different concerns in my opinion. For example, some languages (like X10) allow concrete classes to define a predicate.